src/handler/target.rs

```diff
@@ -320,14 +320,16 @@ impl Target {
                 );
             }
             TargetInit::InitializingPage(cmds) => {
+                let emulation = self.emulation_manager.init_commands(&self.config.viewport);
```

Comment:
Following up with my comments on the `EmulationManager::init_commands` comments, this is where we should check whether the `config.viewport` is present and based on that either queue in the commands or advance the state directly to `Initialized`.

This should probably look something like this

```rust
if let Some(viewport) = self.config.viewport.as_ref() {
    advance_state!(self, cx, now, cmds, self.emulation_manager.init_commands(viewport)); 
} else {
    advance_state!(self, cx, now, cmds, TargetInit::Initialized);
}
```

Reply:
Thanks for the feedback, by the way.